drivers/spi/spi_ll_stm32.c

```diff
@@ -613,6 +613,7 @@ static int transceive(const struct device *dev,
 
 	ret = spi_stm32_configure(dev, config);
 	if (ret) {
+		spi_context_release(&data->ctx, ret);
```

Comment:
As `spi_context_release` is inlined, please use a goto instruction to avoid code increase.
Apply to the other case as well